src/utilities/activities.ts

```diff
@@ -37,13 +37,23 @@ export function activityToPoint(
         }, [])
     : [];
 
+  let activityLabelText = activity.name;
+
+  if (activity.parent_id !== null) {
+    activityLabelText = activity.type;
+  }
+
+  if (activity.unfinished) {
+    activityLabelText = `${activityLabelText} (Unfinished)`;
+  }
+
```

Comment:
I generally don't like mutative code, but the non-mutative equivalent is worse IMO

```
const activityLabelText =
  (
    activity.parent_id !== null
    ? activity.type
    : activity.name
  ) + (
    activity.unfinished
    ? ' (Unfinished)'
    : ''
  )
```

Reply:
Yeah same. I generally don't like the mutative style either. I actually wrote out the non-mutative style first and came to a similar conclusion. The logic is harder to follow.

Reply:
If we ever get a `name` field for simulated activities the logic will simplify nicely.

Reply:
I wonder if there is a programming construct in some other language or as a research topic that would help with this kind of thing. I've always been frustrated with building strings (really any sequence-like thing) and how even template strings get nasty when composing multiple parts.